README.md

```diff
@@ -407,6 +407,16 @@ Permission checks and requests resolve into one of these statuses:
 | `RESULTS.GRANTED`     | The permission is granted                                         |
 | `RESULTS.BLOCKED`     | The permission is denied and not requestable anymore              |
 
+### Getting the result of `PERMISSION.IOS.LOCATION_ALWAYS` on iOS 13 and later
```

Comment:
I'm sorry, but this is a no-go for me. A never ending `Promise` could arm a lot of applications, and unfortunately we can't rely on the fact that developers read the documentation before implementing the feature 😕

Reply:
How do you suggest handling it?

Reply:
I don't know yet. But that's precisely why I don't implemented this in the first time.

Reply:
I don't think there's any way to have this permission work with the v2 API without having either a promise that might never resolve.

For v3 we can do something like have this call resolve instantly and then emit an event when the permission is granted, and don't emit any event if the permission was denied -- but v3 will take a while to be released, and for existing v2 users I think it's better to have a solution with a caveat instead of a bug 🤷‍♂️

Reply:
This is not a bug, but more of a missing feature. A Promise that never resolve is a bug.
The first one leads to feature requests, the second to issues.

I'd rather see people relying on `patch-package` and be sure understand what they're doing than providing a broken (because of the current API design) feature.

Note that it's also doable to create a custom PermissionHandler fork without forking the entire repository :)

Reply:
Fair enough -- do you wanna close this PR then?

Reply:
Not really for now, I'm sure it could be possible in some way to return `blocked` in these cases. I just don't have the bandwidth to check this currently (because of https://github.com/zoontek/react-native-permissions/pull/522)

Reply:
@zoontek @tallpants wouldn't it make sense to rely on `applicationDidBecomeActive` (or other app lifecycle method?) instead of (or along with) `didChangeAuthorizationStatus` as when you prompt the always permissions, app isn't on foreground anymore and changes back when the request dialog closes?
So far this seems to be working fine for me
```
- (void)requestWithResolver:(void (^ _Nonnull)(RNPermissionStatus))resolve
....
[_locationManager setDelegate:self];

  [[NSNotificationCenter defaultCenter] addObserver:self
                                        selector:@selector(handleAppStateDidChange:)
                                        name:UIApplicationDidBecomeActiveNotification
                                        object:nil];
...
```

```
- (void)updatePermission {
    if(_resolve == nil || _reject == nil) return;
    [self checkWithResolver:_resolve rejecter:_reject];
    [_locationManager setDelegate:nil];
    _resolve = nil;
    _reject = nil;
    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIApplicationDidBecomeActiveNotification object:nil];
}

- (void)locationManager:(CLLocationManager *)manager didChangeAuthorizationStatus:(CLAuthorizationStatus)status {
    if (status != kCLAuthorizationStatusNotDetermined && status != kCLAuthorizationStatusAuthorizedWhenInUse) {
        [self updatePermission];
    }
}

- (void)handleAppStateDidChange:(NSNotification *)notification {
    if([notification.name isEqualToString:UIApplicationDidBecomeActiveNotification]) {
        [self updatePermission];
    }
}
```

Reply:
> This is not a bug, but more of a missing feature. A Promise that never resolve is a bug.
> The first one leads to feature requests, the second to issues.
> 
> I'd rather see people relying on `patch-package` and be sure understand what they're doing than providing a broken (because of the current API design) feature.
> 
> Note that it's also doable to create a custom PermissionHandler fork without forking the entire repository :)

How would I go about only forking the PermissionHandler instead of the entire repository? Thanks!

Reply:
@antonigiske

> How would I go about only forking the PermissionHandler instead of the entire repository? Thanks!

I wouldn't recommend that. I believe what you want is https://github.com/ds300/patch-package